lib/pbench/cli/agent/commands/triggers/list.py

```diff
@@ -0,0 +1,80 @@
+"""
+pbench-list-triggers
+
+This script will list all tool triggers from all tool groups, or list tool
+triggers from a specific group.
+
+"""
+import sys
+
+import click
+
+from pbench.agent.utils import setup_logging
+from pbench.cli.agent import CliContext, pass_cli_context
+from pbench.cli.agent.commands.triggers.base import TriggerCommand
+from pbench.cli.agent.options import common_options
+
+
+class TriggerList(TriggerCommand):
+    """List register triggers"""
+
+    def __init__(self, context):
+        super().__init__(context)
+
+        self.logger = setup_logging()
+
+    def execute(self):
+        if not self.pbench_run.exists():
+            # Silently exit if we don't have a pbench run directory.
+            return 1
+
+        if not self.context.group:
+            # list tool triggers in all groups
+            if len(self.groups) == 0:
+                self.logger.error("%s: error fetching list of tool groups", self.name)
+                return 1
+
+            for group in self.groups:
+                trigger = self.gen_tools_group_dir(group) / "__trigger__"
+                if trigger.exists():
+                    with open(trigger) as f:
+                        data = f.read().splitlines()
+                    print("%s: %s" % (group, "  ".join(data)))
```

Comment:
I thought I saw this `splitlines`/`join` go away in the list-triggers PR? Now they're merged and it's back?

If `__trigger__` is a single line with `<start>:<stop>` then it probably doesn't matter. And if it somehow gets multiple lines your `" ".join` should give a similar effect to `$(cat ...)` but ... it still just feels funny. Mostly I guess I'm wondering if this was a deliberate reversion or an accidental regression on the merge.

Reply:
It needs to be rebased, just focus on the register.